api/src/migrations/main/1687301343450-public-notice-post-date.js

```diff
@@ -0,0 +1,31 @@
+const { MigrationInterface, QueryRunner } = require("typeorm");
+
+module.exports = class PublicNoticePostDate1687301343450 {
+
+    async up(queryRunner) {
+        console.log('Starting public_notice (add new column - post_date) migration');
+        await queryRunner.query(`
+    
+        -- add new column - post_date, nullable (for existing data before new column exists)
+            ALTER TABLE app_fom.public_notice ADD COLUMN post_date date;
+
+        -- comment on column - post_date
+            COMMENT ON COLUMN app_fom.public_notice.post_date IS
+            'Date planed for online public notice posted (visible to public) before the FOM reaches the commenting open period';
+
+        -- set post_date value to app_fom.project.commenting_open_date for the past data as default value.
+            update app_fom.public_notice pn set post_date = (
```

Comment:
so for existing public notices, set the post date to commenting open. Maybe we should exclude projects in state INITIAL?

Reply:
Hi Basil, I have added this exclusion in script. However, github section on your review seem a bit strange and does not reflect the change. Please go to the file to verify again for this script please. Thank you Basil.